src/transforms/stack.js

```diff
@@ -81,6 +81,9 @@ function stack(x, y = () => 1, ky, {offset, order, reverse, ...options} = {}) {
         }
         if (offset) offset(stacks, Y1, Y2, Z);
       }
+      if (offset === offsetWiggle || offset === offsetSilhouette) {
+        offsetCenterFacets(Y1, Y2, facets);
+      }
```

Comment:
It’s a bit of a bummer that we special-case the logic for these two offset implementations. Philosophically, none of the built-in methods (e.g., for offsets and orders when stacking, or reducers when binning or grouping) should have “privileged” abilities that go beyond what can be done with a passed-in function. The idea being as much as possible we want people to be able to pass in arbitrary functions that can do as much as the built-in ones.

So, I wonder if that would be possible here? One idea that comes to mind is a separate “offsetCenter” boolean option that defaults to true iff the *offset* option is wiggle or silhouette. But, maybe there’s another way of doing it.

Reply:
The offset logic needs either:
a) each of the stacks, individually, for expand and silhouette
b) all the stacks in one facet, for wiggle
The center logic needs c) all the stacks in all the facets.

Since c > b > a, we can create a reference of all the stacks in all facets and pass them at once to the offset function. This makes writing an offset function a tiny bit more complex (you receive an array of arrays of stacks).

I'm favoring this over having another option.